vncviewer/MonitorIndicesParameter.cxx

```diff
@@ -0,0 +1,209 @@
+#include <limits>
+#include <set>
+#include <stdlib.h>
+#include <stdexcept>
+
+#include "i18n.h"
+#include <FL/Fl.H>
+#include <rfb/LogWriter.h>
+
+#include "MonitorIndicesParameter.h"
+
+using namespace rfb;
+static LogWriter vlog("MonitorIndicesParameter");
+
+MonitorIndicesParameter::MonitorIndicesParameter(const char* name_, const char* desc_, const char* v)
+: StringParameter(name_, desc_, v) {}
+
+std::set<int> MonitorIndicesParameter::getParam()
+{
+    bool valid = false;
+    std::set<int> indices;
+    std::set<int> config_indices;
+    std::vector<MonitorIndicesParameter::Monitor> monitors = this->monitors();
+
+    if (monitors.size() <= 0) {
+        vlog.error(_("Failed to get monitors."));
+        return indices;
+    }
+
+    valid = parse_indices(value, config_indices);
+    if (!valid) {
+        return indices;
+    }
+
+    if (config_indices.size() <= 0) {
+        return indices;
+    }
+
+    // Go through the monitors and see what indices are present in the config.
+    for (int i = 0; i < ((int) monitors.size()); i++) {
+        if (std::find(config_indices.begin(), config_indices.end(), i) != config_indices.end())
+            indices.insert(monitors[i].fltk_index);
+    }
+
+    return indices;
+}
+
+bool MonitorIndicesParameter::setParam(const char* value)
+{
+    std::set<int> indices;
+
+    if (strlen(value) <= 0)
+        return false;
+
+    if (!parse_indices(value, indices)) {
+        vlog.error(_("Parsing failed for FullScreenSelectedMonitors."));
+        return false;
+    }
+
+    return StringParameter::setParam(value);
+}
+
+
+bool MonitorIndicesParameter::setParam(std::set<int> indices)
+{
+    static const int BUF_MAX_LEN = 1024;
+    char buf[BUF_MAX_LEN] = {0};
+    std::set<int> config_indices;
+    std::vector<MonitorIndicesParameter::Monitor> monitors = this->monitors();
+
+    if (monitors.size() <=  0) {
+        vlog.error(_("Failed to get monitors."));
+        // Don't return, store the configuration anyways.
+    }
+
+    for (int i = 0; i < ((int) monitors.size()); i++) {
+        if (std::find(indices.begin(), indices.end(), monitors[i].fltk_index) != indices.end())
+            config_indices.insert(i);
+    }
+
+    int bytes_written = 0;
+    char const * separator = "";
+
+    for (std::set<int>::iterator index = config_indices.begin();
+         index != config_indices.end();
+         index++)
+    {
+        bytes_written += snprintf(
+            buf+bytes_written,
+            BUF_MAX_LEN-bytes_written,
+            "%s%u",
+            separator,
+            (*index)+1
+        );
+
+        separator = ",";
+    }
+
+    return setParam(buf);
+}
+
+static bool parse_number(std::string number, std::set<int>& indices)
+{
+    if (number.size() <= 0)
+        return false;
+
+    int v = strtol(number.c_str(), NULL, 0);
+
+    if (v < 0 || v > std::numeric_limits<int>::max()) {
+        vlog.error(_("The given index (%s) is too large to be valid."), number.c_str());
```

Comment:
The reader of this doesn't know the context of this message, so you need to specify that it is a screen index.